### PR TITLE
AEA-1196 Add oracle skills unit tests

### PR DESCRIPTION
**Pull request by jrriehl**
## Proposed changes

Add unit tests of the oracle data collection and contract interaction skills

## Types of changes

What types of changes does your code introduce to agents-aea?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

_Put an `x` in the boxes that apply._

- [x] I have read the [CONTRIBUTING](../CONTRIBUTING.md) doc
- [x] I am making a pull request against the `develop` branch (left side). Also you should start your branch off our `develop`.
- [x] Lint and unit tests pass locally with my changes
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have checked that code coverage does not decrease.
- [x] I have checked that the documentation about the `aea cli` tool works
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: tests/test_packages/test_skills/test_coin_price/test_behaviours.py [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+        cls.coin_price_behaviour = cast(
+            CoinPriceBehaviour, cls._skill.skill_context.behaviours.coin_price_behaviour
+        )
+
+    def test_send_http_request_message(self):
+        """Test the send_http_request_message method of the coin_price behaviour."""
+        self.coin_price_behaviour.send_http_request_message("GET", "some_url")
+        self.assert_quantity_in_outbox(1)
+        msg = cast(HttpMessage, self.get_message_from_outbox())
+        assert msg, "Wrong message type"
```
Comment (5A11) [inline tests/test_packages/test_skills/test_coin_price/test_behaviours.py:51]: I don't think this line is generally needed. If there is/isn't a message the `self.assert_quantity_in_outbox` line will verify that, unless I am mistaken you wouldn't get `None` in the outbox anyway.

File: tests/test_packages/test_skills/test_coin_price/test_behaviours.py [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+        assert (
+            msg.performative == HttpMessage.Performative.REQUEST
+        ), "Wrong message performative"
+        assert msg.url == "some_url", "Wrong url"
```
Comment (5A11) [inline tests/test_packages/test_skills/test_coin_price/test_behaviours.py:55]: Suggestion, you can generally combine all the attributes you are expecting from your message via `message_has_attributes()`, then whatever attribute you want checked you provide to this method as arguments. E.g.
```
has_attributes, error_str = self.message_has_attributes(
            actual_message=msg,
            message_type= HttpMessage,
            performative= HttpMessage.Performative.REQUEST,
            to=...,
            sender=...,
            url=""some_url"",
        )
        assert has_attributes, error_str
```

File: tests/test_packages/test_skills/test_coin_price/test_behaviours.py [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+        self.assert_quantity_in_outbox(1)
+        msg = cast(PrometheusMessage, self.get_message_from_outbox())
+        assert msg, "Wrong message type"
+        assert (
+            msg.performative == PrometheusMessage.Performative.ADD_METRIC
+        ), "Wrong message performative"
+        assert msg.type == "Gauge", "Wrong metric type"
+        assert msg.title == "some_metric", "Wrong metric title"
+        assert msg.description == "some_description", "Wrong metric description"
+        assert msg.labels == {"label_key": "label_value"}, "Wrong labels"
```
Comment (5A11) [inline tests/test_packages/test_skills/test_coin_price/test_behaviours.py:71]: Similarly here lines 64-71 can be combined into a single call. The `message_has_attributes` method also takes care of the appropriate error message which is returned alongside the result bool.

File: tests/test_packages/test_skills/test_coin_price/test_handlers.py [hunk 4]
```diff
@@ -0,0 +1,240 @@
...
+
+        # handle message
+        self.http_handler.handle(incoming_message)
+
+        # check that data was correctly entered into shared state
+        oracle_data = {"value": 10000000, "decimals": self.coin_price_model.decimals}
+        assert self.http_handler.context.shared_state["oracle_data"] == oracle_data
+
+        # check that outbox contains update_prometheus metric message
+        self.assert_quantity_in_outbox(1)
```
Comment (5A11) [inline tests/test_packages/test_skills/test_coin_price/test_handlers.py:99]: You have many occurrence of just checking the number of messages in the outbox without checking the message attributes. Make sure in all these case, there isn't an attribute that is key for the message to have that haven't checked for.

File: tests/test_packages/test_skills/test_fetch_beacon/test_behaviours.py [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+    def test_send_http_request_message(self):
+        """Test the send_http_request_message method of the fetch_beacon behaviour."""
+        self.fetch_beacon_behaviour.send_http_request_message("GET", "some_url")
+        self.assert_quantity_in_outbox(1)
+        msg = cast(HttpMessage, self.get_message_from_outbox())
+        assert msg, "Wrong message type"
+        assert (
+            msg.performative == HttpMessage.Performative.REQUEST
+        ), "Wrong message performative"
+        assert msg.url == "some_url", "Wrong url"
```
Comment (5A11) [inline tests/test_packages/test_skills/test_fetch_beacon/test_behaviours.py:55]: Similar to above

File: tests/test_packages/test_skills/test_fetch_beacon/test_behaviours.py [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+    def test_act(self):
+        """Test that the act method of the fetch_beacon behaviour puts the correct message in the outbox."""
+        self.fetch_beacon_behaviour.act()
+        self.assert_quantity_in_outbox(1)
+        msg = cast(HttpMessage, self.get_message_from_outbox())
+        assert msg, "Wrong message type"
+        assert (
+            msg.Performative == HttpMessage.Performative
+        ), "Wrong message performative"
+        assert msg.url == self.fetch_beacon_behaviour.beacon_url, "Wrong url"
```
Comment (5A11) [inline tests/test_packages/test_skills/test_fetch_beacon/test_behaviours.py:71]: Similar to above (I won't repeat this comment from here)

File: tests/test_packages/test_skills/test_simple_oracle/test_behaviours.py [hunk 7]
```diff
@@ -0,0 +1,309 @@
...
+        assert msg, "Wrong message type"
+        assert (
+            msg.performative == PrometheusMessage.Performative.ADD_METRIC
+        ), "Wrong message performative"
+        assert msg.type == "Gauge", "Wrong metric type"
+        assert msg.title == "num_oracle_updates", "Wrong metric title"
+        assert (
+            msg.description == "Number of updates published to oracle contract"
+        ), "Wrong metric description"
+        assert msg.labels == {}, "Wrong labels"
```
Comment (5A11) [inline tests/test_packages/test_skills/test_simple_oracle/test_behaviours.py:89]: combined attribute checking is especially apt here :)

**Review by 5A11 — APPROVED** [review]
A couple of suggestions for more clear code, otherwise lgtm

File: tests/test_packages/test_skills_integration/test_coin_price.py [hunk 8]
```diff
@@ -89,7 +89,7 @@ def test_coin_price(self):
         self.run_install()
 
         process = self.run_agent()
-        is_running = self.is_running(process)
+        is_running = self.is_running(process, timeout=30)
```
Comment (DavidMinarsch) [inline tests/test_packages/test_skills_integration/test_coin_price.py:92]: why so long?

Comment (jrriehl) [reply tests/test_packages/test_skills_integration/test_coin_price.py]: This was a check for what seems to be a flaky test, but it looks fine now so I reverted back to default timeout.

**Review by DavidMinarsch — APPROVED** [review]
LGTM

**Pull request merged**